docs/guidance/rayclient-nginx-ingress.md

```diff
@@ -0,0 +1,129 @@
+```sh
+kubectl apply -f https://raw.githubusercontent.com/ray-project/kuberay/master/ray-operator/config/samples/ray-cluster.tls.yaml
+```
+Refer to the [TLS document](tls.md) for more detail.
+
+## Step 5: Create an ingress for the Ray client service
+With the Raycluster running, create an ingress for the Ray client backend service using the [rayclient-ingress](../../ray-operator/config/samples/ingress-rayclient-tls.yaml) example below:
+```sh
+cat << EOF | kubectl apply -f -
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: rayclient-ingress
+  annotations:
+    kubernetes.io/ingress.class: nginx
+    nginx.ingress.kubernetes.io/ssl-passthrough: "true"
+    nginx.ingress.kubernetes.io/backend-protocol: "GRPC"
+spec:
+  rules:
+    - host: "localhost"
+      http:
+        paths:
+        - path: "/"
+          pathType: Prefix
+          backend:
+            service:
+              name: raycluster-tls-head-svc
+              port:
+                number: 10001
+EOF
+```
+The annotation, `nginx.ingress.kubernetes.io/backend-protocol: "GRPC"` sets up the appropriate NGINX configuration to route http/2 traffic to a GRPC backend service. The `nginx.ingress.kubernetes.io/ssl-passthrough: "true"` annotation tells the ingress to forward the encrypted traffic to the backend service to be handled inside the Raycluster itself.
+
+## Step 6: Connecting to Ray client service via the ingress
+Since the Raycluster uses TLS, the local Ray client would require a set of certificates to connect to Raycluster.
+> Warning: Ray client has some known [limitations](https://docs.ray.io/en/latest/cluster/running-applications/job-submission/ray-client.html#things-to-know) and is not actively maintained.
+```sh
+# Download the ca key pair and create a cert signing request (CSR)
+kubectl get secret ca-tls -o template='{{index .data "ca.key"}}'|base64 -d > ./ca.key
+kubectl get secret ca-tls -o template='{{index .data "ca.crt"}}'|base64 -d > ./ca.crt
+openssl req -nodes -newkey rsa:2048 -keyout ./tls.key -out ./tls.csr -subj '/CN=local'
+cat <<EOF >./cert.conf
+authorityKeyIdentifier=keyid,issuer
+basicConstraints=CA:FALSE
+subjectAltName = @alt_names
+[alt_names]
+DNS.1 = localhost
+IP.1 = 127.0.0.1
+EOF
+# Sign and create a tls cert
+openssl x509 -req -CA ./ca.crt -CAkey ./ca.key -in ./tls.csr -out ./tls.crt -days 365 -CAcreateserial -extfile ./cert.conf
+
+# Connect Ray client to the Raycluster using the tls keypair and the ca cert
+python -c '
+import os
+import ray
+os.environ["RAY_USE_TLS"] = "1"
+os.environ["RAY_TLS_SERVER_CERT"] = os.path.join("./", "tls.crt")
+os.environ["RAY_TLS_SERVER_KEY"] = os.path.join("./", "tls.key")
+os.environ["RAY_TLS_CA_CERT"] = os.path.join("./", "ca.crt")
+ray.init(address="ray://localhost", logging_level="DEBUG")'
+```
+
+The output should be similar to:
+```
+2023-04-25 16:33:32,452	INFO client_builder.py:253 -- Passing the following kwargs to ray.init() on the server: logging_level
```

Comment:
I cannot reproduce this. 

```python
# test.py
import os
import ray
os.environ["RAY_USE_TLS"] = "1"
os.environ["RAY_TLS_SERVER_CERT"] = os.path.join("./", "tls.crt")
os.environ["RAY_TLS_SERVER_KEY"] = os.path.join("./", "tls.key")
os.environ["RAY_TLS_CA_CERT"] = os.path.join("./", "ca.crt")

ray.init(address="ray://localhost", logging_level="DEBUG")

print(ray.cluster_resources())
```

<img width="1091" alt="Screen Shot 2023-05-08 at 2 07 04 PM" src="https://user-images.githubusercontent.com/20109646/236935310-602240a9-e068-48f1-adbb-d26dd3c31211.png">

Reply:
I used `rayproject/ray:2.4.0-py310` in the TLS RayCluster. 

In addition, my local environment is `Python 3.10.10` and Ray 2.4.0.
<img width="614" alt="Screen Shot 2023-05-08 at 2 10 17 PM" src="https://user-images.githubusercontent.com/20109646/236935856-75ca165a-c117-4959-88a3-8e4166202312.png">